src/thyra/ThyraAccount.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  IAccountBalanceResponse,
+  IAccountDetails,
+  IAccountSignResponse,
+} from '..';
+import { IAccount } from '../account/IAccount';
+import { JsonRpcResponseData, getRequest, postRequest } from './RequestHandler';
+import { THYRA_ACCOUNTS_URL } from './ThyraProvider';
+
+const THYRA_BALANCE_URL = `https://my.massa/massa/addresses?attributes=balance&addresses`;
+
+interface ISignOperation {
+  operation: string;
+  batch?: boolean;
+  correlationId?: string;
+}
+
+interface IBalance {
+  final: string;
+  pending: string;
+}
+
+interface IAddressesBalances {
+  addressesAttributes: {
+    [key: string]: { balance: IBalance };
+  };
+}
+
+export class ThyraAccount implements IAccount {
+  private _providerName: string;
+  private _address: string;
+  private _name: string;
+
+  public constructor({ address, name }: IAccountDetails, providerName: string) {
+    this._address = address;
+    this._name = name;
+    this._providerName = providerName;
+  }
+
+  public address(): string {
+    return this._address;
+  }
+
+  public name(): string {
+    return this._name;
+  }
+
+  public providerName(): string {
+    return this._providerName;
+  }
+
+  public async balance(): Promise<IAccountBalanceResponse> {
+    let signOpResponse: JsonRpcResponseData<IAddressesBalances> = null;
+    try {
+      signOpResponse = await getRequest<IAddressesBalances>(
+        `${THYRA_BALANCE_URL}=${this._address}`,
+      );
+    } catch (ex) {
+      console.error(`Thyra account balance error`);
+      throw ex;
+    }
+    if (signOpResponse.isError || signOpResponse.error) {
+      throw signOpResponse.error;
+    }
+    const balance: IBalance =
+      signOpResponse.result.addressesAttributes[this._address].balance;
+    return {
+      balance: balance.final,
+    };
+  }
+
+  public async sign(data: Uint8Array | string): Promise<IAccountSignResponse> {
```

Comment:
from the user point of view, it could be great to have a description of the data to sign to display to the user along side with the SIGN button of the wallet. this will require to modify the standard https://github.com/massalabs/massa-standards/blob/main/wallet/dapps-communication.md?plain=1#L359

Reply:
I'm not sure to understand. If you want to convert the binary part to the real data, this should be possible in the furtur when we will use protoBuff.
If you want to duplicate the data, one version in clear and one version encoded, how do you plan to be sure that both versions are consistent?

Reply:
@SlnPons help me ^^ 

no need to assure consistency, it's just a tooltip for the user if I had well understood

Reply:
@gregLibert @Thykof I don't understand your discussion. I'm sorry.
Indeed, I'm currently trying to collect the information we can retrieve from the blockchain about tx and its description / any related info to improve the user experience. However, binaries, protoBuff, duplicate data sound not relevant for this 1st version of the wallet. KISS.

Reply:
@gregLibert @Thykof I think the MR in its current state is good to go  - it has been tested, I have added Thyra tests to the dapp too. Maybe we could merge it now ?